test/common/upstream/eds_test.cc

```diff
@@ -1430,6 +1431,108 @@ TEST_F(EdsTest, MalformedIP) {
                             "setting cluster type to 'STRICT_DNS' or 'LOGICAL_DNS'");
 }
 
+class EdsAssignmentTimeoutTest : public EdsTest {
+public:
+  EdsAssignmentTimeoutTest() : EdsTest(), interval_timer_(nullptr) {
+    EXPECT_CALL(dispatcher_, createTimer_(_))
+        .WillOnce(Invoke([this](Event::TimerCb cb) {
+          timer_cb_ = cb;
+          EXPECT_EQ(nullptr, interval_timer_);
+          interval_timer_ = new Event::MockTimer();
+          return interval_timer_;
+        }))
+        .WillRepeatedly(Invoke([](Event::TimerCb) { return new Event::MockTimer(); }));
```

Comment:
This was needed to allow you to avoid recreating the timer on each EDS update? How come?

Reply:
createTimer was getting called twice, once in eds and second one I was not able to trace.
This allows me to associate interval_time_ to the first call, which I know is for eds.